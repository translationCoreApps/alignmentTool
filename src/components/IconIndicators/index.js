import React from 'react';
import PropTypes from 'prop-types';
import { Glyphicon } from 'react-bootstrap';

const IconIndicators = ({
  translate,
  verseEditStateSet,
  verseEditIconEnable,
  commentStateSet,
  commentIconEnable,
  bookmarkStateSet,
  bookmarkIconEnable,
  verseEditClickAction,
  commentClickAction,
  bookmarkClickAction,
}) => {
  return (
    <div style={{ display: 'flex', justifyContent: 'flex-end' }}>
      {verseEditIconEnable ? (
        <Glyphicon
          glyph="pencil"
          style={{
            margin: '0px 20px',
            color: verseEditStateSet ? 'var(--highlight-color)' : 'var(--reverse-color)'
          }}
          title={verseEditStateSet ? translate('icons.verse_edits_found') : translate('icons.no_verse_edits_found')}
          onClick={verseEditClickAction}
        />
      ) : '' }
      {commentIconEnable ? (
        <Glyphicon
          glyph="comment"
          style={{
            margin: '0px 20px',
            color: commentStateSet ? 'var(--highlight-color)' : 'var(--reverse-color)'
          }}
          title={commentStateSet ? translate('icons.comments_found') : translate('icons.no_comments_found')}
          onClick={commentClickAction}
        />
      ) : '' }
      {bookmarkIconEnable ? (
        <Glyphicon
          glyph="bookmark"
          style={{
            margin: '0px 20px',
            color: bookmarkStateSet ? 'var(--highlight-color)' : 'var(--reverse-color)'
          }}
          title={bookmarkStateSet ? translate('icons.bookmarked') : translate('icons.not_bookmarked')}
          onClick={bookmarkClickAction}
        />
      ) : '' }
    </div>
  );
};

IconIndicators.propTypes = {
  translate: PropTypes.func.isRequired,
  verseEditStateSet: PropTypes.bool.isRequired,
  verseEditIconEnable: PropTypes.bool.isRequired,
  commentStateSet: PropTypes.bool.isRequired,
  commentIconEnable: PropTypes.bool.isRequired,
  bookmarkStateSet: PropTypes.bool.isRequired,
  bookmarkIconEnable: PropTypes.bool.isRequired,
  verseEditClickAction: PropTypes.func.isRequired,
  bookmarkClickAction: PropTypes.func.isRequired,
  commentClickAction: PropTypes.func.isRequired,
};

IconIndicators.defaultProps = {
  verseEditStateSet: false,
  verseEditIconEnable: false,
  commentStateSet: false,
  commentIconEnable: false,
  bookmarkStateSet: false,
  bookmarkIconEnable: false,
  verseEditClickAction: () => false,
  bookmarkClickAction: () => false,
  commentClickAction: () => false,
};

export default IconIndicators;
