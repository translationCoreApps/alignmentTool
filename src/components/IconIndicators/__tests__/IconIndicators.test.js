/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import IconIndicators from '../../IconIndicators/index';

describe('IconIndicators component:', () => {

  test('Icons not visible', () => {
    // given
    const visible = false;
    const set = false;
    const props = getBaseProperties({}, visible, set);

    // when
    const component = renderer.create(
      <IconIndicators {...props} />
    );

    // then
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Icons visible and not set', () => {
    // given
    const visible = true;
    const set = false;
    const props = getBaseProperties({}, visible, set);

    // when
    const component = renderer.create(
      <IconIndicators {...props} />
    );

    // then
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Icons visible and set', () => {
    // given
    const visible = true;
    const set = true;
    const props = getBaseProperties({}, visible, set);

    // when
    const component = renderer.create(
      <IconIndicators {...props} />
    );

    // then
    expect(component.toJSON()).toMatchSnapshot();
  });
});

//
// helpers
//

function getBaseProperties(props, visible, set) {
  return {
    ...props,
    translate: text => text,
    verseEditStateSet: set,
    verseEditIconEnable: visible,
    commentStateSet: set,
    commentIconEnable: visible,
    bookmarkStateSet: set,
    bookmarkIconEnable: visible,
    toolsSettings: {}
  };
}

