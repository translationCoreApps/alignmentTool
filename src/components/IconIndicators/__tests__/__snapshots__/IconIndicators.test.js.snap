// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconIndicators component: Icons not visible 1`] = `
<div
  style={
    Object {
      "alignItems": "baseline",
      "display": "flex",
      "justifyContent": "flex-end",
    }
  }
>
  
  
  
  <span
    className="glyphicon glyphicon-option-vertical"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "18px",
        "margin": "0 0 0 10px",
        "padding": 0,
      }
    }
    title={undefined}
  />
</div>
`;

exports[`IconIndicators component: Icons visible and not set 1`] = `
<div
  style={
    Object {
      "alignItems": "baseline",
      "display": "flex",
      "justifyContent": "flex-end",
    }
  }
>
  <span
    className="glyphicon glyphicon-pencil"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--reverse-color)",
        "margin": "0px 20px",
      }
    }
    title="icons.no_verse_edits_found"
  />
  <span
    className="glyphicon glyphicon-comment"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--reverse-color)",
        "margin": "0px 20px",
      }
    }
    title="icons.no_comments_found"
  />
  <span
    className="glyphicon glyphicon-bookmark"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--reverse-color)",
        "margin": "0px 20px",
      }
    }
    title="icons.not_bookmarked"
  />
  <span
    className="glyphicon glyphicon-option-vertical"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "18px",
        "margin": "0 0 0 10px",
        "padding": 0,
      }
    }
    title={undefined}
  />
</div>
`;

exports[`IconIndicators component: Icons visible and set 1`] = `
<div
  style={
    Object {
      "alignItems": "baseline",
      "display": "flex",
      "justifyContent": "flex-end",
    }
  }
>
  <span
    className="glyphicon glyphicon-pencil"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--highlight-color)",
        "margin": "0px 20px",
      }
    }
    title="icons.verse_edits_found"
  />
  <span
    className="glyphicon glyphicon-comment"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--highlight-color)",
        "margin": "0px 20px",
      }
    }
    title="icons.comments_found"
  />
  <span
    className="glyphicon glyphicon-bookmark"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--highlight-color)",
        "margin": "0px 20px",
      }
    }
    title="icons.bookmarked"
  />
  <span
    className="glyphicon glyphicon-option-vertical"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "18px",
        "margin": "0 0 0 10px",
        "padding": 0,
      }
    }
    title={undefined}
  />
</div>
`;
